integrationtest/backend/elasticsearch/src/test/java/org/hibernate/search/integrationtest/backend/elasticsearch/schema/management/ElasticsearchIndexSchemaManagerUpdateCustomSettingsIT.java

```diff
@@ -0,0 +1,216 @@
+				.isEqualTo( "\"3\"" );
+		assertThat( elasticsearchClient.index( index.name() ).settings( "index.number_of_replicas" ).get() )
+				.isEqualTo( "\"3\"" );
+	}
+
+	@Test
+	public void change_analysis() {
+		elasticsearchClient.index( index.name() ).deleteAndCreate( "index",
+				" { " +
+				"   'number_of_shards': '3', " +
+				"   'number_of_replicas': '3', " +
+				"   'analysis': { " +
+				"     'analyzer': { " +
+				"       'my_standard-english': { " +
+				"         'type': 'standard', " +
+				"         'stopwords': '_english_' " +
+				"       }, " +
+				"       'my_analyzer_ngram': { " +
+				"         'type': 'custom', " +
+				"         'tokenizer': 'my_analyzer_ngram_tokenizer' " +
+				"       } " +
+				"     }, " +
+				"     'tokenizer': { " +
+				"       'my_analyzer_ngram_tokenizer': { " +
+				"         'type': 'ngram', " +
+				"         'min_gram': '2', " +
+				"         'max_gram': '3' " +
+				"       } " +
+				"     } " +
+				"   } " +
+				" } "
+		);
+
+		setupAndUpdateIndex();
+
+		assertJsonEquals(
+				" { " +
+				" 	'analyzer': { " +
+				" 		'my_standard-english': { " +
+				" 			'type': 'standard', " +
+				" 			'stopwords': '_english_' " +
+				" 		}, " +
+				" 		'my_analyzer_ngram': { " +
+				" 			'type': 'custom', " +
+				" 			'tokenizer': 'my_analyzer_ngram_tokenizer' " +
+				" 		} " +
+				" 	}, " +
+				" 	'tokenizer': { " +
+				" 		'my_analyzer_ngram_tokenizer': { " +
+				" 			'type': 'ngram', " +
+				" 			'min_gram': '5', " +
+				" 			'max_gram': '6' " +
+				" 		} " +
+				" 	} " +
+				" } ",
+				elasticsearchClient.index( index.name() ).settings( "index.analysis" ).get()
+		);
+
+		assertThat( elasticsearchClient.index( index.name() ).settings( "index.number_of_shards" ).get() )
+				.isEqualTo( "\"3\"" );
+		assertThat( elasticsearchClient.index( index.name() ).settings( "index.number_of_replicas" ).get() )
+				.isEqualTo( "\"3\"" );
+	}
+
+	@Test
+	public void change_numberOfShards() {
+		elasticsearchClient.index( index.name() ).deleteAndCreate( "index",
+				" { " +
+				"   'number_of_shards': '7', " +
+				"   'number_of_replicas': '3', " +
+				"   'analysis': { " +
+				"     'analyzer': { " +
+				"       'my_standard-english': { " +
+				"         'type': 'standard', " +
+				"         'stopwords': '_english_' " +
+				"       }, " +
+				"       'my_analyzer_ngram': { " +
+				"         'type': 'custom', " +
+				"         'tokenizer': 'my_analyzer_ngram_tokenizer' " +
+				"       } " +
+				"     }, " +
+				"     'tokenizer': { " +
+				"       'my_analyzer_ngram_tokenizer': { " +
+				"         'type': 'ngram', " +
+				"         'min_gram': '2', " +
+				"         'max_gram': '3' " +
+				"       } " +
+				"     } " +
+				"   } " +
+				" } "
+		);
+
+		assertThatThrownBy( () -> setupAndUpdateIndex() )
+				.isInstanceOf( SearchException.class )
+				.hasMessageContainingAll(
+						"Unable to update settings", "setting [index.number_of_shards], not updateable" );
```

Comment:
I'd be surprised if all Elasticsearch versions returned the same error message. Did you run a full build?

Reply:
You're right. Testing the Elasticsearch generated error text is not a good idea in case, let's remove this assertion.